quodlibet/browsers/covergrid/models.py

```diff
@@ -0,0 +1,254 @@
+# Copyright 2022 Thomas Leberbauer
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+from gi.repository import GObject, Gio
+
+from quodlibet import _, ngettext, app
+from quodlibet.qltk.models import ObjectModelSort, ObjectStore, ObjectModelFilter
+from quodlibet.util.library import background_filter
+
+
+class AlbumListItem(GObject.Object):
+
+    def __init__(self, album=None):
+        super().__init__()
+        self._album = album
+        self._cover = None
+        self._label = None
+
+        self.connect('notify::album', self._album_changed)
+
+    def load_cover(self, size, cancelable=None):
+        def callback(cover):
+            self._cover = cover
+            self.notify('cover')
+
+        manager = app.cover_manager
+        # Skips this during testing
+        if manager:
+            manager.get_pixbuf_many_async(
+                self._album.songs, size, size, cancelable, callback)
+
+    def format_label(self, pattern):
+        self._label = pattern % self._album
+        self.notify('label')
+
+    def _album_changed(self, model, prop):
+        self._label = None
+
+    @GObject.Property
+    def album(self):
+        return self._album
+
+    @GObject.Property
+    def cover(self):
+        return self._cover
+
+    @GObject.Property
+    def label(self):
+        return self._label
+
+
+class AlbumListCountItem(AlbumListItem):
+
+    def load_cover(self, *args, **kwargs):
+        self.notify('cover')
+
+    def format_label(self, pattern=None):
+        n = self.__n_albums
+        self._label = ('<b>%s</b>\n' +
```

Comment:
This is quite confusing to read for me; why not just substitute the _All Albums_ immediately?

Also why are we ignoring `pattern`?

Reply:
I don't quite understand. Like this?
```python
self._label = (('<b>%s</b>\n' % _('All Albums')) +
    (ngettext('%d album', '%d albums', n) % n))
```
I guess I just went for doing it in one substitution instead of two.

Your second point: `AlbumListCountItem` inherits from `AlbumListItem` and overrides `format_label`. So `pattern` is included to match the method signature from the base class, but the "All albums" entry doesn't actually use it.

Reply:
Yes, like that! To me it's easier to read (easier still if a variable is extracted, or maybe if f-strings work with gettext, but not sure)

>Your second point: AlbumListCountItem inherits from AlbumListItem and overrides format_label. So pattern is included to match the method signature from the base class, but the "All albums" entry doesn't actually use it.

Ah right, yes that one doesn't need the pattern at all, thanks